pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -177,5 +180,16 @@ private static void createLeadControllerResourceIfNeeded(String helixClusterName
 
       helixAdmin.addResource(helixClusterName, LEAD_CONTROLLER_RESOURCE_NAME, idealState);
     }
+
+    // Create resource config for lead controller resource if it doesn't exist
+    ResourceConfig resourceConfig = configAccessor.getResourceConfig(helixClusterName, LEAD_CONTROLLER_RESOURCE_NAME);
+    if (resourceConfig == null) {
+      resourceConfig = new ResourceConfig(LEAD_CONTROLLER_RESOURCE_NAME);
+    }
+    // Set RESOURCE_ENABLED to false if it's absent in resource config
+    if (resourceConfig.getSimpleConfig("RESOURCE_ENABLED") == null) {
```

Comment:
putIfAbsent?

Reply:
I'd love to but `ResouceConfig` isn't a hashmap and there's no such method for now. :)

Reply:
Make sense lol